alpha/apps/kaltura/lib/v2Tov7Utils.class.php

```diff
@@ -0,0 +1,138 @@
+	{
+		KalturaLog::log("Searching for " . $v2PluginName . " " . strlen($v2PluginName));
+		$translation = self::v2toV7PluginMap();
+		if(isset($translation[$v2PluginName]))
+		{
+			$v7PluginInfo = $translation[$v2PluginName];
+			KalturaLog::log("Found " . print_r($v7PluginInfo, true));
+			return $v7PluginInfo;
+		}
+		else
+		{
+			throw new Exception ($v2PluginName);
+		}
+	}
+
+	static function addV2toV7plugins($flashvars, &$bundleConfig, &$playerConfig)
+	{
+		if(!$flashvars)
+		{
+			return;
+		}
+		$unHandledPlugins = array();
+
+		//Merge v7 config
+		foreach ($flashvars as $key => $value)
+		{
+			if(self::isVarPlugin($key))
+			{
+				//get plugin name
+				KalturaLog::log("V2 to V7 adding plugin: " . $key. " value:" . $value);
+				if ($value) {
+					if (!$bundleConfig) {
+						$bundleConfig = [];
+					}
+					$key = trim(trim($key, '"'), "'");
+					$v2PluginName = explode(".", $key);
+					try
+					{
+						$v7PluginName = self::getV7PluginInfo($v2PluginName[0]);
+						$bundleConfig = array_merge($bundleConfig, [$v7PluginName[0] => "{latest}"]);
+						$v7PluginConfig = $v7PluginName[1];
+						if (!isset($playerConfig->plugins)) {
+							$playerConfig->plugins = new stdClass();
+						}
+						if (!isset($playerConfig->plugins->$v7PluginConfig)) {
+							$playerConfig->plugins->$v7PluginConfig = new stdClass();
+						}
+					}
+					catch (Exception $e){
+						$unHandledPlugins[] = $e->getMessage();
+					}
+				}
+			}
+		}
+		if(sizeof($unHandledPlugins))
+		{
+			throw new Exception ("Unhandled plugins: " . implode(", ", $unHandledPlugins));
+		}
+	}
+
+	static function addV2toV7config($flashvars, $uiconfId)
+	{
+		$config = [];
+		//Merge v7 config
+
+		if($flashvars)
+		{   foreach ($flashvars as $key => $value) {
+				$key = trim(trim($key, '"'), "'");
+				$config[$key] = json_decode($value);
+			}
+		}
+        $config["uiconf_id"] = $uiconfId;
+		return $config;
+	}
+
+	private static function isVarPlugin($varKeyName)
+	{
+		return str_contains($varKeyName, ".plugin");
+	}
+
+	private static function v2toV7PluginMap()
+	{
+		return
+			["info" => ["playkitscreen", "playkit-js-info"],
+			"quiz"          => ["playkit-ivq", "ivq"],
+			"moderation"    => ["playkit-moderation", "playkit-js-moderation"],
+			"playlistAPI"   => ["playkit-playlist", "playlist"],
+			"liveStatus"    => ["playkit-kaltura-live","kaltura-live"],
+			"related"       => ["playkit-related", "related"],
+			"dualScreen"    => ["playkit-dual-screen", "dualscreen"],
+			"video360"      => ["playkit-vr" ,"vr"],
+			"raptMedia"     => ["rapt", "rapt"],
+			"transcript"    => ["playkit-transcript", "playkit-js-transcript"],
+			"qna"           => ["playkit-qna", "qna"],
+			"bumper"        => [ "playkit-bumper" , "bumper" ],
+			"infoScreen" => ["playkit-info", "playkit-js-info"]];
+	}
+
+	public static function getBundledFacade()
+	{
+		//build key based on version
+		$facadeVersion = kConf::getArrayValue('v2tov7FacadeVersion','playkit-js');
```

Comment:
Where is the config for v2tov7FacadeVersion do you have another PR for it?

Reply:
This needs to go to local.ini